src/app.tsx

```diff
@@ -9,24 +9,30 @@ import { darkTheme } from './styles/themes/dark-theme';
 import { ThemeContext } from './core/theme/theme-contex';
 import { loadTranslations } from './core/i18n';
 import { persistStore } from 'redux-persist';
+import { SplashScreen } from './components/splash-screen/SplashScreen';
+
 const AppContainer = createAppContainer(RootNavigation);
 
 const store = configureStore();
 const persistor = persistStore(store);
+const MIN_TIME_ANIMATION = 2;
 
 interface IState {
     appReady: boolean;
+    defaultTimer: any;
 }
 
 export default class App extends React.Component<{}, IState> {
+    public interval: any = null;
     //    public unsubscribe =
     private translationsLoaded: boolean = false;
     private reduxStateLoaded: boolean = false;
 
     constructor(props: any) {
         super(props);
         this.state = {
-            appReady: false
+            appReady: false,
+            defaultTimer: 1
```

Comment:
Let's keep it simple.
You should use `setTimeout()` not `setInterval()`. In state keep a property `splashAnimationDone: boolean`.

Example: 

```ts
constructor(props: any) {
  this.state = {
    appReady: false,
    splashAnimationDone: false
  }
  setTimeout(() => this.setState({splashAnimationDone: true}); // you can move this in componentDidMount()
}

render() {
  if (this.state.appReady && this.state.splashAnimationDone) {
    // all done, render app
  } else {
    // render splash
  }
}
```